src/api/libcellml/analyserequationast.h

```diff
@@ -0,0 +1,354 @@
+     * types:
+     *  - Assignment type:
+     *     - ASSIGNMENT: the assignment operator
+     *  - Relational and logical operators:
+     *    - EQ: the equal to function;
+     *    - NEQ: the not equal to function;
+     *    - LT: the less than function;
+     *    - LEQ: the less than or equal to function;
+     *    - GT: the greater than function;
+     *    - GEQ: the greater than or equal to function;
+     *    - AND: the and function;
+     *    - OR: the or function
+     *    - XOR: the exclusive or function;
+     *    - NOT: the not function;
+     *  - Arithmetic operators:
+     *    - PLUS: the plus operator;
+     *    - MINUS: the minus operator;
+     *    - TIMES: the times operator;
+     *    - DIVIDE: the divide operator;
+     *    - POWER: the power operator;
+     *    - ROOT: the root operator;
+     *    - ABS: the absolute value function;
+     *    - EXP: the exponential function;
+     *    - LN: the natural logarithm function;
+     *    - LOG: the common logarithm function;
+     *    - CEILING: the ceiling function;
+     *    - FLOOR: the floor function;
+     *    - MIN: the minimum function;
+     *    - MAX: the maximum function;
+     *    - REM: the remainder function;
+     *  - Calculus elements:
+     *    - DIFF: the differentiation operator;
+     *  - Trigonometric operators:
+     *    - SIN: the sine function;
+     *    - COS: the cosine function;
+     *    - TAN: the tangent function;
+     *    - SEC: the secant function;
+     *    - CSC: the cosecant function;
+     *    - COT: the cotangent function;
+     *    - SINH: the hyperbolic sine function;
+     *    - COSH: the hyperbolic cosine function;
+     *    - TANH: the hyperbolic tangent function;
+     *    - SECH: the hyperbolic secant function;
+     *    - CSCH: the hyperbolic cosecant function;
+     *    - COTH: the hyperbolic cotangent function;
+     *    - ASIN: the arc sine function;
+     *    - ACOS: the arc cosine function;
+     *    - ATAN: the arc tangent function;
+     *    - ASEC: the arc secant function;
+     *    - ACSC: the arc cosecant function;
+     *    - ACOT: the arc cotangent function;
+     *    - ASINH: the arc hyperbolic sine function;
+     *    - ACOSH: the arc hyperbolic cosine function;
+     *    - ATANH: the arc hyperbolic tangent function;
+     *    - ASECH: the arc hyperbolic secant function;
+     *    - ACSCH: the arc hyperbolic cosecant function;
+     *    - ACOTH: the arc hyperbolic cotangent function;
+     *  - Piecewise statement:
+     *    - PIECEWISE: the "piecewise" statement;
+     *    - PIECE: the "piece" part of a "piecewise" statement;
+     *    - OTHERWISE: the "otherwise" part of a "piecewise" statement;
+     *  - Token elements:
+     *    - CI: an identifier (i.e. the name of a model variable);
+     *    - CN: a number;
+     *  - Qualifier elements:
+     *    - DEGREE: the degree (or order) of a differential equation;
```

Comment:
Yes, our analyser only supports differential equations of the first order and when we create the AST for such an equation, we don't generate the `DEGREE` information since it's always one. However, `DEGREE` is also used for the `ROOT` function and this is where it is used in an AST. So, `DEGREE` is needed, but its documentation is wrong and has been corrected in commit 66027ba5.